app/views/spaces/index/_space_facility_listing.html.erb

```diff
@@ -1,5 +1,6 @@
 <ul class="grid grid-cols-2 gap-2 mt-2">
   <% filtered_facilities.each do |facility| %>
+    <% next unless space.relevant_facilities.map{|sf|sf.facility}.include?(facility) %>
```

Comment:
This thing feels like it should have been done somewhere else, and probably not recursively for each filtered_facility for each space in the search results...

Reply:
Definitely could move this part out `space.relevant_facilities.map{|sf|sf.facility}` and just have `.include?(facility)` here

Reply:
Definitely could move this part out space.relevant_facilities.map{|sf|sf.facility} and just have .include?(facility) here

Reply:
Will try!

Reply:
Better now? @mathias234